src/safeposix/syscalls/fs_calls.rs

```diff
@@ -26,158 +59,225 @@ impl Cage {
+                                // and is only used when the file exists and is a regular file
+                                if O_TRUNC == (flags & O_TRUNC) {
+                                    // Close the existing file object and remove it from the FileObject Hashtable using the inodenumber
+                                    let entry = FILEOBJECTTABLE.entry(inodenum);
+                                    if let interface::RustHashEntry::Occupied(occ) = &entry {
+                                        occ.get().close().unwrap();
+                                    }
 
-                if mode & (S_IRWXA | S_FILETYPEFLAGS as u32) != mode {
-                    return syscall_error(Errno::EPERM, "open", "Mode bits were not sane");
-                } //assert sane mode bits
+                                    f.size = 0;
 
-                let time = interface::timestamp(); //We do a real timestamp now
-                let newinode = Inode::File(GenericInode {
-                    size: 0,
-                    uid: DEFAULT_UID,
-                    gid: DEFAULT_GID,
-                    mode: effective_mode,
-                    linkcount: 1,
-                    refcount: 1,
-                    atime: time,
-                    ctime: time,
-                    mtime: time,
-                });
+                                    // Update the timestamps as well
+                                    let latest_time = interface::timestamp(); 
+                                    f.ctime = latest_time;
+                                    f.mtime = latest_time; 
 
-                let newinodenum = FS_METADATA
-                    .nextinode
-                    .fetch_add(1, interface::RustAtomicOrdering::Relaxed); //fetch_add returns the previous value, which is the inode number we want
-                if let Inode::Dir(ref mut ind) =
-                    *(FS_METADATA.inodetable.get_mut(&pardirinode).unwrap())
-                {
-                    ind.filename_to_inode_dict.insert(filename, newinodenum);
-                    ind.linkcount += 1;
-                    //insert a reference to the file in the parent directory
-                } else {
-                    return syscall_error(
-                        Errno::ENOTDIR,
-                        "open",
-                        "tried to create a file as a child of something that isn't a directory",
-                    );
-                }
-                FS_METADATA.inodetable.insert(newinodenum, newinode);
-                log_metadata(&FS_METADATA, pardirinode);
-                log_metadata(&FS_METADATA, newinodenum);
+                                    // Remove the previous file and add a new one of 0 length
+                                    if let interface::RustHashEntry::Occupied(occ) = entry {
+                                        occ.remove_entry();
+                                    }
 
```

Comment:
remove this extra whitespace

Reply:
I checked, and there is not any extra whitespace here; it is just because of the indentation as the code section has moved inside the matching block.